docs/java-rest/high-level/migration.asciidoc

```diff
@@ -0,0 +1,137 @@
+[[java-rest-high-level-migration]]
+== Migration Guide
+
+This section describes how to migrate existing code from the `TransportClient`
+to the new Java High Level REST Client released with the version 5.6.0
+of Elasticsearch.
+
+=== Motivations around a new Java client
+
+The existing `TransportClient` has always been part of Elasticsearch since https://github
+.com/elastic/elasticsearch/blob/b3337c312765e51cec7bde5883bbc0a08f56fb65/modules/elasticsearch/src/main/java/org/elasticsearch/client
+/transport/TransportClient.java[its very first commit]. Although it covers the whole set of Elasticsearch APIs,
+it always has had various flaws:
+
+- it is tightly coupled with Elasticsearch and uses Elasticsearch's internal implementations
+- it uses Java serialization to transport objects over the wire using their binary representations
+
+These two aspects forces any Java application that communicates with an Elasticsearch cluster to use
+a `TransportClient` in a version compatible with the remote cluster's version. That's perfect most of
```

Comment:
I agree - I changed _perfect_ for _acceptable_. Let me know if you think the whole sentence should be reworded.